src/tracksdata/graph/_base_graph.py

```diff
@@ -1092,6 +1092,82 @@ def bbox_spatial_filter(
 
         return BBoxSpatialFilter(self, frame_attr_key=frame_attr_key, bbox_attr_key=bbox_attr_key)
 
+    @abc.abstractmethod
+    def assign_track_ids(
+        self,
+        output_key: str = DEFAULT_ATTR_KEYS.TRACK_ID,
+        reset: bool = True,
+        track_id_offset: int | None = None,
+        node_ids: list[int] | None = None,
+    ) -> rx.PyDiGraph:
+        """
+        Compute and assign track ids to nodes.
+        Parameters
+        ----------
+        output_key : str
+            The key of the output track id attribute.
+        reset : bool
+            Whether to reset the track ids of the graph. If True, the track ids will be reset to -1.
+        track_id_offset : int | None
+            The starting track id, useful when assigning track ids to a subgraph.
+            If None, the track ids will start from 1 or from the maximum existing track id + 1
+            if the output_key already exists and reset is False.
+        node_ids : list[int] | None
+            The node ids to assign track ids to. If None, all nodes are used.
+
+        Returns
+        -------
+        rx.PyDiGraph
+            A compressed graph (parent -> child) with track ids lineage relationships.
+            If node_ids is provided, it will only include linages including those nodes.
+        """
+        raise NotImplementedError(f"{self.__class__.__name__} backend does not support track id assignment.")
+
+    # Internal utility to compute the exact set of nodes to assign when a subset is requested
+    def _compute_track_node_ids(self, seeds: list[int] | None) -> list[int]:
```

Comment:
I really like this function, I can see some use cases besides updating `track_ids`
Do you think we could rename and remove the `_` prefix to make it "public"?

I'm not really good at naming functions. I thought of `non_branching_connected_nodes` or `non_branching_connected_components`, but I'm open to your suggestions.